src/js/core.js

```diff
@@ -44,16 +44,17 @@ module.exports.getUrlParameter = (name) => {
 
 module.exports.parseHTML = parseHTML;
 
-module.exports.renderTemplate = (id, context = {}) => {
+function renderTemplate (id, context = {}) {
```

Comment:
As a stylistic thing, I would prefer that function declarations go to the bottom of the file and all of the modules.exports statements to stay grouped together around the middle. Nothing by any means which should block a merge, but just a change I may introduce later in a cleanup commit if we can agree on that point.

Reply:
I hear you. My preference is to export methods right after defining them. If I don't see a `module.export` after a method, I get the impression that it doesn't need to be exported.

They get hoisted in the end anyways :)

This is why we use styleguides (hint [AirBnB JS Styleguide](https://github.com/airbnb/javascript#modules))

That being said, feel free to introduce this change in the future.

Lastly, with regards to `core.cfg.pathnames`, if we are using these values directly in the html, wouldn't it be beneficial to have the "page names/title" in there as well? That way we can just have a `core` method generate navs rather than having the below duplicated in our pages.

```
<nav class="navbar navbar-expand-md navbar-dark bg-dark">
      <button class="navbar-toggler navbar-toggler-right" type="button" data-toggle="collapse" data-target="#navbarNavAltMarkup" aria-controls="navbarNavAltMarkup" aria-expanded="false" aria-label="Toggle navigation">
        <span class="navbar-toggler-icon"></span>
      </button>
      <a class="navbar-brand" href="#">Leverage</a>
      <div class="collapse navbar-collapse" id="navbarNavAltMarkup">
        <div class="navbar-nav">
          <a class="nav-item nav-link" href="/">Home <span class="sr-only">(current)</span></a>
          <a class="nav-item nav-link active" href="#">Candidates</a>
          <a class="nav-item nav-link" href="#">Races</a>
        </div>
      </div>
    </nav>
```

Reply:
> My preference is to export methods right after defining them. If I don't see a module.export after a method, I get the impression that it doesn't need to be exported.

That's fine with me. I don't actually have a functional reason for preferring it the other way. Like we always say, as long as we're consistent.

> My preference is to export methods right after defining them. If I don't see a module.export after a method, I get the impression that it doesn't need to be exported.

Don't get me wrong.. the structure of the config is going through some growing pains. It could definitely benefit from a reassessment.

> That way we can just have a core method generate navs rather than having the below duplicated in our pages.

It seems to me that because of the way we've chosen to do templating, we're going to end up with one of the following three situations regardless of how we choose to solve this problem:

1. Duplicating the nav template between pages
2. Inconsistency in how we maintain templates

Since the duplicate problem could easily become more prevalent down the road, my *preference* would be to re-solve how we do templating in general, before we get to much deeper down the rabbit hole.

Reply:
> Inconsistency in how we maintain templates

This should be easily fixed with having a set guideline to follow with regards to naming conventions et al.

> Duplicating the nav template between pages.

Agreed, we need to address now.

With regards to our current templating method, I'm open to better alternatives. Thoughts?

I was thinking of [Partials](https://github.com/janl/mustache.js) earlier but, that could easily become another pile of mess.

Reply:
<img width="328" alt="multipass" src="https://user-images.githubusercontent.com/3523537/31971026-2b9691f4-b8e8-11e7-8fae-c15c3846a209.png">